### PR TITLE
Add DeferredEventsTrait

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Tests pass?   | ✔️
| Fixed issues  | #36

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -158,12 +198,18 @@ The package tests are checked with [Infection](https://infection.github.io/) mut
 
 ### Static analysis
 
-The code is statically analyzed with [Phan](https://github.com/phan/phan/wiki). To run static analysis:
+The code is statically analyzed with [Psalm](https://psalm.dev/). To run static analysis:
```
Comment (rustamwin) [inline README.md:201]: This package still uses phan static analysis.

Comment (vjik) [reply README.md:201]: Yes, phan. Why you change in readme phan to psalm ?

Comment (vjik) [reply README.md:201]: I make another PR (https://github.com/yiisoft/event-dispatcher/pull/42) which change phan to psalm.

Comment (yiiliveext) [reply README.md:201]: Maybe this naming will be better 
`recordEvent` => `pushEvent`
`releaseEvents` => `getEvents`

Comment (vjik) [reply README.md:201]: > `recordEvent` => `pushEvent`
`push` - sending somewhere, but here save to yourself

> `releaseEvents` => `getEvents`

`get` - only get, but here get and clear

**Comment by viktorprogger** [comment]
WDYT about making a service instead of a trait? It should be more flexible as I see:

```php
interface DeferredEventService {
    public function pushEvent(object $event): void;
    public function releaseEvents(): void;
}
```

**Pull request closed without merging**